### ISSUE TITLE
an HTTPProxy with no routes, includes, or tcpproxy is marked as valid

### ISSUE THREAD

**Issue by skriss** [projectcontour/contour#2835]
**What steps did you take and what happened:**
Create the following `HTTPProxy`:
```yaml
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: empty
spec:
  virtualhost:
    fqdn: foo.bar.com
```

See that the proxy is marked as valid:
```bash
NAME    FQDN          TLS SECRET   STATUS   STATUS DESCRIPTION
empty   foo.bar.com                valid    valid HTTPProxy
```

**What did you expect to happen:**
The `HTTPProxy` should be marked invalid since it's "empty" and likely is a result of user error.

**Anything else you would like to add:**
We should also add either a kubebuilder validation or a Gatekeeper validation for this to give the user immediate feedback.

**Environment:**

- Contour version: `v1.7.0`

**Comment by skriss** [comment]
xref #2141

**Comment by skriss** [comment]
#2141 discusses more complex issues related to inclusion and chains of proxies. However, I think we could fix this more constrained issue (#2835) independent of that. I *believe* that any HTTPProxy that does not directly define at least one route, include, or tcpproxy can universally be considered invalid.

**Comment by youngnick** [comment]
In the inclusion case, I think that should mean that, if _all_ of the includes HTTPProxies also contain nothing, then the containing proxy will contain nothing, and therefore be invalid by this rule. I think that's okay, but it does mean that the listener could be removed if there's only one (invalid) HTTPProxy.
